src/Query/Group.ts

```diff
@@ -151,13 +162,44 @@ export class Group {
         return [Group.escapeMarkdownCharacters(filename)];
     }
 
-    private static groupByRoot(task: Task): string[] {
-        const path = task.path.replace(/\\/g, '/');
-        const separatorIndex = path.indexOf('/');
-        if (separatorIndex == -1) {
-            return ['/'];
+    private static groupByFn(task: Task, arg?: GroupingArg): string[] {
+        const paramsArgs: [string, any][] = [
+            ['description', task.description],
+            ['done', task.doneDate],
+            ['due', task.dueDate],
+            ['filename', task.filename],
+            ['happens', new HappensDateField().earliestDate(task)],
+            ['header', task.precedingHeader],
+            ['markdown', task.originalMarkdown],
+            ['path', task.path.replace('.md', '')],
+            ['priority', task.priority],
+            ['recurrence', task.recurrence],
+            ['root', Group.root(task)],
+            ['scheduled', task.scheduledDate],
+            ['start', task.startDate],
+            ['status', task.status],
+            ['t', task],
+            ['tags', task.tags],
+            ['task', task],
+            ['urgency', task.urgency],
+        ];
+
+        const params = paramsArgs.map(([p]) => p);
+        const groupBy = arg && new Function(...params, `return ${arg}`);
+
+        if (groupBy instanceof Function) {
```

Comment:
How might it not be an instance of function at this point?

Reply:
I think if `arg?` is nullish `const groupBy = arg && new Function..` would make `groupBy` nullish as well